pkg/comp-functions/functions/vshnredis/pvcresize.go

```diff
@@ -0,0 +1,276 @@
+	}
+	stsUpdated := stsSize == desiredSize
+
+	deletionJob := &batchv1.Job{}
+	if observedJob {
+		err := json.Unmarshal(xJob.Status.AtProvider.Manifest.Raw, deletionJob)
+		if err != nil {
+			return runtime.NewFatalErr(ctx, "cannot unmarshal sts deleter job", err)
+		}
+	}
+
+	// The job hasn't been observed yet, so we need to keep it in desired, or we will have a recreate loop
+	// Also as long as it hasn't finished we need to make sure it exists.
+	if (!observedJob || deletionJob.Status.Succeeded < 1) || (sts.Status.ReadyReplicas == 0 && !stsUpdated) {
+		err := addDeletionJob(ctx, iof, comp)
+		if err != nil {
+			return runtime.NewFatalErr(ctx, "cannot create RBAC for the deletion job", err)
+		}
+	}
+
+	return runtime.NewNormal()
+}
+
+func needReleasePatch(ctx context.Context, comp *vshnv1.VSHNRedis, values map[string]interface{}) (bool, runtime.Result) {
+	releaseSizeValue, found, err := unstructured.NestedString(values, "master", "persistence", "size")
+	if !found {
+		return false, runtime.NewFatalErr(ctx, "could not find disk size in release", fmt.Errorf("disk size not found in release"))
+	}
+	if err != nil {
+		return false, runtime.NewFatalErr(ctx, "failed to read size from release", err)
+	}
+
+	desiredInt, err := getSizeAsInt(comp.Spec.Parameters.Size.Disk)
+	if err != nil {
+		return false, runtime.NewFatalErr(ctx, "cannot parse desired disk size", err)
+	}
+
+	releaseInt, err := getSizeAsInt(releaseSizeValue)
+	if err != nil {
+		return false, runtime.NewFatalErr(ctx, "cannot parse release disk size", err)
+	}
+
+	return desiredInt > releaseInt, nil
+
+}
+
+func getSizeAsInt(size string) (int64, error) {
+	parsedSize, err := resource.ParseQuantity(size)
+	if err != nil {
+		return 0, err
+	}
+	finalSize, _ := parsedSize.AsInt64()
+	return finalSize, nil
+}
+
+func getObservedOrDesiredRelease(ctx context.Context, iof *runtime.Runtime) (*helmv1beta1.Release, error) {
+	r := &helmv1beta1.Release{}
+	err := iof.Observed.Get(ctx, r, redisRelease)
+	if err != nil && err != runtime.ErrNotFound {
+		return nil, fmt.Errorf("cannot get redis release from observed iof: %v", err)
+	}
+
+	if err == runtime.ErrNotFound {
+		err := iof.Desired.Get(ctx, r, redisRelease)
+		if err != nil {
+			return nil, fmt.Errorf("cannot get redis release from desired iof: %v", err)
+		}
+	}
+	return r, nil
+}
+
+func addDeletionJob(ctx context.Context, iof *runtime.Runtime, comp *vshnv1.VSHNRedis) error {
+	ns := iof.Config.Data["controlNamespace"]
+
+	job := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      comp.Name + "-sts-deletion-job",
+			Namespace: ns,
+		},
+		Spec: batchv1.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					ServiceAccountName: "sa-sts-deleter",
+					RestartPolicy:      corev1.RestartPolicyNever,
+					Containers: []corev1.Container{
+						{
+							Name:            "sts-deleter",
+							Image:           "bitnami/kubectl",
```

Comment:
is it always available image in cluster? I hit some issues in other project when image was not present and cluster could only use some trusted images

Reply:
This seems like a very specific case. Should I document in the readme that it depends on this image?

Reply:
yeah, should be fine